File: scaffold/src/main/java/top/xuqingquan/web/DownloadListener.java
package top.xuqingquan.web;

import android.net.Uri;
import androidx.annotation.MainThread;

public class DownloadListener {

    @MainThread
    public boolean onStart(String url, String userAgent, String contentDisposition, String mimetype, long contentLength, Extra extra) {
        return false;
    }

    public void onProgress(String url, long downloaded, long length, long usedTime) {
    }

    @MainThread
    public boolean onResult(Throwable throwable, Uri path, String url, Extra extra) {
        return false;
    }
}

File: scaffold/src/main/java/top/xuqingquan/web/DownloadListener.kt
package top.xuqingquan.web

import android.net.Uri
import androidx.annotation.MainThread

open class DownloadListener {

    @MainThread
    open fun onStart(
        url: String,
        userAgent: String,
        contentDisposition: String,
        mimetype: String,
        contentLength: Long,
        extra: Extra
    ): Boolean {
        return false
    }

    open fun onProgress(url: String, downloaded: Long, length: Long, usedTime: Long) {}

    @MainThread
    open fun onResult(throwable: Throwable, path: Uri, url: String, extra: Extra): Boolean {
        return false
    }
}
